x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/external/http/retry/ErrorResponse.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.inference.external.http.retry;
+
+import java.util.Objects;
+
+public class ErrorResponse {
+
+    // Denotes an error object that was not found
+    public static final ErrorResponse UNDEFINED_ERROR = new ErrorResponse(false);
```

Comment:
The reason I need this is to differentiate between an error object being present but a specific field in the object missing (for example `message`) and when no error exists. Previously we were returning null in both of those scenarios. Now if there's an exception or the main field that defines an error object doesn't exist we'll return this instance to indicate that there was no error present.

If a nested field is missing we return a created instance with `errorStructureFound` set to true.